FileSystemManager/Controllers/FileSystemController.cs

```diff
@@ -346,98 +357,31 @@ public IActionResult RenameItem(string path, string newName)
                 {
                     return NotFound();
                 }
+                // Normalize the path for database search
+                var normalizedPath = decodedPath.Replace('\\', '/');
+                // Fetch all records and filter in memory
+                var fileMetadataList = _context.FileMetadata.ToList();
+                var fileMetadata = fileMetadataList.SingleOrDefault(fm => fm.FilePath.Replace('\\', '/') == normalizedPath);
+                if (fileMetadata == null)
+                {
+                    return NotFound("File metadata not found in the database");
+                }
+                // Update file metadata with new name and path
+                var newFilePath = Path.Combine(Path.GetDirectoryName(normalizedPath), newName).Replace('\\', '/');
+                fileMetadata.FileName = newName;
+                fileMetadata.FilePath = newFilePath;
+                fileMetadata.DateModified = DateTime.Now;
+                _context.FileMetadata.Update(fileMetadata);
+                _context.SaveChanges();
 
                 return Ok();
             }
-            catch
+            catch (Exception ex)
```

Comment:
Is there no rollback in case there is an exception?

Reply:
More specifically, code block 

```
if (Directory.Exists(fullPath))
                {
                    Directory.Move(fullPath, newFullPath);
                }
                else if (System.IO.File.Exists(fullPath))
                {
                    System.IO.File.Move(fullPath, newFullPath);
                }
```
Will run and change the filename but db call might fail

Reply:
https://github.com/FrostyMo/FileSystemManager/assets/160615896/fb2e907e-068e-4a1a-8664-86c6e9d59bf8

Please upload the new fine and rename it